workflow/templateresolution/context.go

```diff
@@ -281,9 +283,16 @@ func (ctx *Context) WithWorkflowTemplate(name string) (*Context, error) {
 
 // WithWorkflowTemplate creates new context with a wfv1.TemplateHolder.
 func (ctx *Context) WithClusterWorkflowTemplate(name string) (*Context, error) {
-	cwftmpl, err := ctx.cwftmplGetter.Get(name)
+	cwftmpl, err := ctx.GetClusterWorkflowTemplate(name)
 	if err != nil {
 		return nil, err
 	}
 	return ctx.WithTemplateBase(cwftmpl), nil
 }
+
+func (ctx *Context) GetClusterWorkflowTemplate(name string) (*wfv1.ClusterWorkflowTemplate, error) {
+	if ctx.cwftmplGetter == nil && reflect.ValueOf(ctx.cwftmplGetter).IsNil() {
+		return nil, errors.Errorf(errors.CodeBadRequest, "Cannot access clusterworkflowtemplates resource")
```

Comment:
I don't think we should return HTTP style errors this deep in the code, normal error should suffice